### PR TITLE
nickle: update to 2.99

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Update

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: math/nickle/Portfile [hunk 1]
```diff
@@ -17,14 +18,41 @@ long_description    Nickle is a programming language based prototyping environme
...
+# math test. autotools buils cannot generate tables:
+# fmt: illegal option -- -
+# meson build generates broken tables:
+# 11/22 math.5c FAIL 1.18s exit status 1
+# math-tables.5c: syntax error before "exp"
+# It is not enough to replace invalid --width with -width
+# or alternatives, since lines with . are not handled
+# correctly anyway. So unfortunately, we need to depend
+# on coreutils here, which can handle linuxisms.
+patchfiles-append   patch-fix-broken-tests-syntax.diff
```
Comment (reneeotten) [inline math/nickle/Portfile:37]: please open an upstream issue for them to resolve this; it's not something MacPorts wants to patch forever

Comment (barracuda156) [reply math/nickle/Portfile:37]: I did not find an upstream repo where I can submit a patch.

This is my port, by the way. Please let me maintain it. And I am pretty sure that no one gonna touch it for years, If I do not do that, so the problem is not with patching forever here.

Comment (reneeotten) [reply math/nickle/Portfile:37]: (1) latest upstream version is `2.101` ; please check if some stuff has been resolved there already.  (2) Upstream appears active so there must be a way to reach them.

**Comment by reneeotten** [comment]
feel free to re-open the PR once you have fixed things

**Pull request closed without merging**